### PR TITLE
docs: Add common use cases

### PR DESCRIPTION
**Pull request by jamilbk**
Adds common use cases of Firezone.

### CODE REVIEW

**Comment by jamilbk** [comment]
Adding @ReactorScram for another set of eyes -- 10 minutes of review should be more than enough 🙏

File: website/public/images/kb/use-cases/nat-gateway/verify.png
Comment (ReactorScram) [inline website/public/images/kb/use-cases/nat-gateway/verify.png:1]: Hope ifconfig.net is chill with this lol

File: website/src/app/kb/use-cases/host-access/readme.mdx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import Link from "next/link";
+import PlanBadge from "@/components/PlanBadge";
+
+<PlanBadge plans={["starter", "team", "enterprise"]}>
+
+# Access a Remote Host
+
+</PlanBadge>
+
+In this guide, we'll be using Firezone to secure access to a host based on the
```
Comment (ReactorScram) [inline website/src/app/kb/use-cases/host-access/readme.mdx]: nit
```suggestion
In this guide, we'll use Firezone to secure access to a host based on the
```

File: website/src/app/kb/use-cases/nat-gateway/readme.mdx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+single, static IP address to use services that require source traffic to come
+from specified IP whitelists.
+
+After completing this guide, your team's traffic will be routed to a Firezone
+Gateway and then out to the internet using its public IP address.
+
+<Alert color="info">
+  See our our [Terraform
+  examples](https://www.github.com/firezone/firezone/tree/main/terrraform/examples)
+  for an high availability example of this guide using Terraform on Google Cloud
```
Comment (ReactorScram) [inline website/src/app/kb/use-cases/nat-gateway/readme.mdx]:
```suggestion
  for a high availability example of this guide using Terraform on Google Cloud
```

File: website/src/app/kb/use-cases/readme.mdx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+<Alert color="info">
+  These are just a few ways Firezone can help your organization achieve zero
+  trust access. If you have a use case you'd like to see covered here, please
+  [let us know](https://github.com/firezone/firezone/issues/new/choose)!
+</Alert>
+
+## Table of Contents
+
+- [Block malicious DNS](/kb/use-cases/secure-dns): Use Firezone to improve your
+  team's Internet security by blocking DNS qeueries to known malicious domains.
```
Comment (ReactorScram) [inline website/src/app/kb/use-cases/readme.mdx]:
```suggestion
  team's Internet security by blocking DNS queries to known malicious domains.
```

File: website/src/app/kb/use-cases/readme.mdx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+- [Scale access to a VPC](/kb/use-cases/scale-vpc-access): Scale access to a
+  Google Cloud VPC using Terraform to deploy multiple Gateways.
+- [Route through a public IP](/kb/use-cases/nat-gateway): Route some your team's
```
Comment (ReactorScram) [inline website/src/app/kb/use-cases/readme.mdx]:
```suggestion
- [Route through a public IP](/kb/use-cases/nat-gateway): Route some of your team's
```

File: website/src/app/kb/use-cases/secure-dns/readme.mdx [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+</Alert>
+
+## Step 2: Add the resolver in Firezone
+
+1. [Sign in](https://app.firezone.dev) to your Firezone account and navigate to
+   `Settings -> DNS` tab.
+1. Add each of the IP addresses of the DNS resolvers you want to use in the
+   **Client DNS** section.
+1. Click **Save** to apply the changes. The Firezone Client will automatically
+   update its configuration to use the new DNS resolvers.
```
Comment (ReactorScram) [inline website/src/app/kb/use-cases/secure-dns/readme.mdx]:
```suggestion
1. Click **Save** to apply the changes. The Firezone Clients will automatically
   update their configuration to use the new DNS resolvers.
```

Nit: plural?

File: website/src/app/kb/use-cases/secure-dns/readme.mdx [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+</Alert>
+
+## Step 3: Verify
+
+You can verify that DNS is being filtered by querying a known malicious domain
+from a device with the Firezone Client signed in and checking if you receive an
+answer of `0.0.0.0` or `NXDOMAIN`.
+
+For example, the `zeustracker.abuse.ch` domain is associated with the Zeus
+botnet and is commonly blocked by many DNS providers:
```
Comment (ReactorScram) [inline website/src/app/kb/use-cases/secure-dns/readme.mdx:83]: That's pretty scary 😨 Should there be an info box "don't actually open that URL in a browser"?

File: website/src/app/kb/use-cases/nat-gateway/readme.mdx [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+
+</PlanBadge>
+
+In this guide, we'll walk through the configuration required to use Firezone to
+route traffic through a public IP address, sometimes known as a NAT Gateway
+configuration.
+
+This is commonly used when you need your team's traffic to appear to come from a
+single, static IP address to use services that require source traffic to come
+from specified IP whitelists.
```
Comment (ReactorScram) [inline website/src/app/kb/use-cases/nat-gateway/readme.mdx]:
```suggestion
from specified IP allowlists.
```
To match Github's precedent and other places in our docs

File: website/src/app/kb/use-cases/nat-gateway/readme.mdx [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+<Alert color="warning">
+  This guide assumes the Gateway has a public IP attached. This is required to
+  function as a NAT gateway.
+</Alert>
+
+## Step 1: Create Resource(s) matching the traffic you want to route
+
+1. In your [admin portal](https://app.firezone.dev), go to `Sites -> <site>` and
+   click the `Add Resource` button.
+1. Add Resource(s) for each service that requires an IP whitelist. For example,
```
Comment (ReactorScram) [inline website/src/app/kb/use-cases/nat-gateway/readme.mdx]:
```suggestion
1. Add Resource(s) for each service that requires an IP allowlist. For example,
```

File: website/src/app/kb/use-cases/readme.mdx [hunk 9]
```diff
@@ -0,0 +1,41 @@
+import SupportOptions from "@/components/SupportOptions";
+import NextStep from "@/components/NextStep";
+import Alert from "@/components/DocsAlert";
+
+# Use Cases
+
+Learn how to solve a variety of common secure access challenges using Firezone.
```
Comment (ReactorScram) [inline website/src/app/kb/use-cases/readme.mdx]:
```suggestion
Learn how to solve a variety of common secure access challenges, by using Firezone.
```
Totally a nit but I kinda read it as "challenges using Firezone" and wanted to break up the brand name from the idea of challenges.

Could also be "solve a variety... with Firezone" or something.

File: website/src/app/kb/use-cases/secure-dns/readme.mdx [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+| Quad9                         | `9.9.9.11`, `149.112.112.11`, `2620:fe::11`, `2620:fe::fe:11`        |
+| Comodo Secure DNS             | `8.26.56.26`, `8.20.247.20`                                          |
+| dns0.eu                       | `193.110.81.9`, `185.253.5.9`                                        |
+| CleanBrowsing Security Filter | `185.228.168.9`, `185.228.169.9`, `2a0d:2a00:1::2`, `2a0d:2a00:2::2` |
+| NextDNS                       | Varies. See [NextDNS](https://nextdns.io/) for more information.     |
+
+<Alert color="warning">
+  Each provider varies greatly in which domains they block which will affect
+  their effectiveness depending on your security requirements. We recommend
+  testing a few providers to see which one works best for you.
```
Comment (ReactorScram) [inline website/src/app/kb/use-cases/secure-dns/readme.mdx]:
```suggestion
  Each provider blocks different domains. We recommend
  testing a few providers to see which one works best for you.
```

File: website/src/app/kb/use-cases/secure-dns/readme.mdx [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+  <Image
+    src="/images/kb/use-cases/secure-dns/step2.png"
+    alt="Add the resolver in Firezone"
+    width={1200}
+    height={1200}
+  />
+</Link>
+
+<Alert color="info">
+  Firezone currently supports upstream resolvers available over UDP/53 only. See
```
Comment (ReactorScram) [inline website/src/app/kb/use-cases/secure-dns/readme.mdx]:
```suggestion
  Firezone only supports upstream resolvers available over UDP/53. See
```

File: website/src/app/kb/use-cases/scale-vpc-access/readme.mdx [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+
+## Step 1: Create a Resource
+
+1. In your [admin portal](https://app.firezone.dev), go to `Sites -> <site>` and
+   click the `Add Resource` button.
+1. Select `CIDR` as the Resource type.
+1. Enter the CIDR range of the VPC you want to secure access to. This should be
+   a range of IPv4 or IPv6 addresses that's directly reachable from the
+   Gateway(s) in your Site.
+1. Name the Resource something descriptive, like `SJC demo net`. You'll refer to
```
Comment (ReactorScram) [inline website/src/app/kb/use-cases/scale-vpc-access/readme.mdx:48]: Who is SJC?

File: website/src/app/kb/use-cases/nat-gateway/readme.mdx [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+
+# Route Traffic Through a Public IP
+
+</PlanBadge>
+
+In this guide, we'll walk through the configuration required to use Firezone to
+route traffic through a public IP address, sometimes known as a NAT Gateway
+configuration.
+
+This is commonly used when you need your team's traffic to appear to come from a
```
Comment (ReactorScram) [inline website/src/app/kb/use-cases/nat-gateway/readme.mdx]:
```suggestion
Use this when you need your team's traffic to appear to come from a
```

File: website/src/app/kb/use-cases/postgres-access/readme.mdx [hunk 14]
```diff
@@ -0,0 +1,76 @@
...
+
+- A Site that contains the Postgres database you want to secure access to.
+  [Create a Site](/kb/deploy/sites) if you haven't already.
+- One or more Gateways deployed within the Site.
+  [Deploy a Gateway](/kb/deploy/gateways) if you don't have any in the Site
+  where this database is located.
+
+<Alert color="warning">
+  For access to high-traffic services like databases, we highly recommend
+  setting up multiple Gateways for load balancing. See [Deploying multiple
```
Comment (ReactorScram) [inline website/src/app/kb/use-cases/postgres-access/readme.mdx]:
```suggestion
  We recommend setting up multiple Gateways 
  for load balancing access to high-traffic services like databases. See [Deploying multiple
```

File: website/src/app/kb/use-cases/saas-app-access/readme.mdx [hunk 15]
```diff
@@ -0,0 +1,78 @@
...
+</Link>
+
+## Step 2: Create a Policy
+
+1. In the `Policies` tab, click the `Add Policy` button.
+1. Create a Policy for the Resource you created in Step (1). Be sure to select
+   the appropriate Group and Resource for the Policy.
+
+## Step 3: Done!
+
```
Comment (ReactorScram) [inline website/src/app/kb/use-cases/saas-app-access/readme.mdx:74]: For a public app like Github, there's no way to be sure that Firezone is actually working properly, unless you do something like open the dev tools and check that you're connecting to 100.96.0.0, right?

With a lot of UX change we could have the GUI Clients light up a resource when it's been recently accessed or something.

Actually the portal has audit logs, right? Should we ask customers to verify those the first time they hook up a new Policy or Client?

File: website/src/app/kb/use-cases/host-access/readme.mdx [hunk 16]
```diff
@@ -0,0 +1,71 @@
...
+This is useful when you have a host that may be behind a firewall that you want
+to keep secure, but still need to access it from external networks like the
+internet.
+
+After completing this guide, you'll be able to access the host using its private
+IP address and any services running on it, like SSH or RDP.
+
+## Prerequisites
+
+- A Site that contains the host you want to secure access to.
```
Comment (ReactorScram) [inline website/src/app/kb/use-cases/host-access/readme.mdx]: Actually the host isn't in that Site until it's created as a Resource in Step 1.

Like, it's logically in the Site, from the customer saying "We have this building, that's a site", but it's not in the Firezone capital-S Site until you're following the instructions.

Maybe "A Site that _will_ contain the host..."

File: website/src/app/kb/use-cases/private-network-access/readme.mdx [hunk 17]
```diff
@@ -0,0 +1,78 @@
...
+to keep secure, but still need to access it from external networks like the
+internet.
+
+<Alert color="warning">
+  In general, we recommend using a more granular approach to secure access using
+  either [DNS](/kb/use-cases/web-app-access) or
+  [IP-based](/kb/use-cases/host-access) Resources instead of the blanket
+  approach used in this guide. We recommend using this guide only if using DNS
+  or IP-based Resources is not feasible, or if you want a stepping stone towards
+  a more granular approach that still works with existing systems today.
```
Comment (ReactorScram) [inline website/src/app/kb/use-cases/private-network-access/readme.mdx]:
```suggestion
  approach used in this guide. Only use this guide if using DNS
  or IP-based Resources is not feasible, or if you need a stepping stone towards a more granular approach.
```
nit

File: website/src/app/kb/use-cases/private-network-access/readme.mdx [hunk 18]
```diff
@@ -0,0 +1,78 @@
...
+</Alert>
+
+## Prerequisites
+
+- A Site that contains the subnet you want to secure access to.
```
Comment (ReactorScram) [inline website/src/app/kb/use-cases/private-network-access/readme.mdx]: Same as above, "Site that _will_ contain" or similar.

File: website/src/app/kb/use-cases/web-app-access/readme.mdx [hunk 19]
```diff
@@ -0,0 +1,87 @@
...
+
+- A Site that contains the web app you want to secure access to.
+  [Create a Site](/kb/deploy/sites) if you haven't already.
+- One or more Gateways deployed within the Site.
+  [Deploy a Gateway](/kb/deploy/gateways) if you don't have any in the Site
+  where this web app is located.
+
+<Alert color="warning">
+  For reliable access to high-traffic web apps, we highly recommend setting up
+  multiple Gateways for load balancing. See [Deploying multiple
```
Comment (ReactorScram) [inline website/src/app/kb/use-cases/web-app-access/readme.mdx]:
```suggestion
  For reliable access to high-traffic web apps, set up
  multiple Gateways for load balancing. See [Deploying multiple
```

**Review by ReactorScram — CHANGES_REQUESTED** [review]
Should the PR title be `docs: Add common use cases`?

Most are nits but some are spelling / grammar errors.

File: website/public/images/kb/use-cases/nat-gateway/verify.png
Comment (jamilbk) [reply website/public/images/kb/use-cases/nat-gateway/verify.png:1]: forgiveness vs permission

File: website/src/app/kb/use-cases/nat-gateway/readme.mdx (hunk 7 above)
Comment (jamilbk) [reply website/src/app/kb/use-cases/nat-gateway/readme.mdx]: 👍

File: website/src/app/kb/use-cases/readme.mdx (hunk 3 above)
Comment (jamilbk) [reply website/src/app/kb/use-cases/readme.mdx]: Weird that codespell didn't catch this

File: website/src/app/kb/use-cases/saas-app-access/readme.mdx (hunk 15 above)
Comment (jamilbk) [reply website/src/app/kb/use-cases/saas-app-access/readme.mdx:74]: If the allowlist is configured in GitHub then it should block access that's not coming from Firezone.

But yeah, we should have a status indicator dot next to each Resource in the clients that is green when an Online gateway exists to serve the Resource.

Perhaps we can grey the Resource out if (a) it's a DNS resource and (b) DNS didn't properly initialize in the Client for our session.

File: website/src/app/kb/use-cases/scale-vpc-access/readme.mdx (hunk 12 above)
Comment (jamilbk) [reply website/src/app/kb/use-cases/scale-vpc-access/readme.mdx:48]: San Jose Airport. Would `US-west` be better here?

File: website/src/app/kb/use-cases/secure-dns/readme.mdx (hunk 6 above)
Comment (jamilbk) [reply website/src/app/kb/use-cases/secure-dns/readme.mdx:83]: It doesn't actually do anything -- it's just a convenient always-blocked domain that was used to track zeus malware C&C queries

**Review by jamilbk — COMMENTED** [review]
Will push a new commit with the changes.

File: website/src/app/kb/use-cases/scale-vpc-access/readme.mdx (hunk 12 above)
Comment (jamilbk) [reply website/src/app/kb/use-cases/scale-vpc-access/readme.mdx:48]: I'll leave it cause I don't want to go and take all those screenshots again 😅

Comment (ReactorScram) [reply website/src/app/kb/use-cases/scale-vpc-access/readme.mdx:48]: Good point. I just wondered if it stood for "Some (J)Giant Company" or some generic placeholder

**Review by ReactorScram — APPROVED** [review]
LGTM

**Pull request merged**